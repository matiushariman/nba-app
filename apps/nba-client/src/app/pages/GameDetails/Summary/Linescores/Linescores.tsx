import LinescoresTable from './LinescoresTable';
import { Card } from '../../../../components/ui';

const Linescores = () => {
  return (
    <Card className="p-5 grid gap-3 divide-y">
      <div>
        <p className="text-2xl font-bold uppercase">Linescores</p>
      </div>
      <div className="pt-5">
        <LinescoresTable />
      </div>
    </Card>
  );
};

export default Linescores;
