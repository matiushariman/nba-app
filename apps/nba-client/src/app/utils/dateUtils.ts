import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
