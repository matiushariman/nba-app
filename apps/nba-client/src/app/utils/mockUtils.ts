export const shouldUseMock = process.env['NODE_ENV'] === 'development';
