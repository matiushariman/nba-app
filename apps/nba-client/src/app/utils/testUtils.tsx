import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
