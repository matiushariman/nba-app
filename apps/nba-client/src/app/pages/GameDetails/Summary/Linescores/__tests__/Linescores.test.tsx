import Linescores from '../Linescores';
import { render } from '../../../../../utils/testUtils';

jest.mock('../../../../../hooks/useGetBoxscore', () => ({
  selectLinescores: jest.fn().mockReturnValueOnce([
    {
      teamTricode: 'LAL',
      score: 92,
      periods: [
        {
          period: 1,
          score: 20,
        },
        {
          period: 2,
          score: 15,
        },
        {
          period: 3,
          score: 25,
        },
        {
          period: 4,
          score: 32,
        },
      ],
    },
    {
      teamTricode: 'BOS',
      score: 131,
      periods: [
        {
          period: 1,
          score: 24,
        },
        {
          period: 2,
          score: 34,
        },
        {
          period: 3,
          score: 31,
        },
        {
          period: 4,
          score: 42,
        },
      ],
    },
  ]),
  selectPeriod: jest.fn().mockReturnValueOnce([
    {
      period: 1,
      label: 'Q1',
    },
    {
      period: 2,
      label: 'Q2',
    },
    {
      period: 3,
      label: 'Q3',
    },
    {
      period: 4,
      label: 'Q4',
    },
  ]),
}));

jest.mock('../../../../../hooks/useGetGame', () => ({
  selectGameLeaders: jest.fn().mockReturnValueOnce({
    homeLeaders: {
      personId: 1629627,
      name: 'Zion Williamson',
      jerseyNum: '1',
      position: 'F',
      teamTricode: 'NOP',
      playerSlug: null,
      points: 35,
      rebounds: 8,
      assists: 3,
    },
    awayLeaders: {
      personId: 1629028,
      name: 'Deandre Ayton',
      jerseyNum: '22',
      position: 'C',
      teamTricode: 'PHX',
      playerSlug: null,
      points: 28,
      rebounds: 12,
      assists: 5,
    },
  }),
}));

describe('pages/GameDetails/Summary/Linescores', () => {
  it('should render Linescores that matches snapshot', () => {
    const { baseElement } = render(<Linescores />);

    expect(baseElement).toMatchSnapshot();
  });
});
