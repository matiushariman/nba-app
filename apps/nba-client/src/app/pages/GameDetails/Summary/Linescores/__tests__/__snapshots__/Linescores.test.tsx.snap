// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/Linescores should render Linescores that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="p-5 grid md:grid-flow-col gap-6 md:divide-x divide-y md:divide-y-0 shadow-md bg-white rounded"
    >
      <div
        class="md:col-span-3"
      >
        <div
          class="grid gap-3 divide-y"
        >
          <div>
            <p
              class="text-2xl font-bold uppercase"
            >
              Linescores
            </p>
          </div>
          <div
            class="pt-5"
          >
            <table
              class="table-auto w-full"
            >
              <thead
                class="bg-gray-300"
              >
                <tr>
                  <th
                    align="left"
                    class="uppercase text-xs pt-1 pb-1 pl-4"
                  >
                    Team
                  </th>
                  <th
                    class="uppercase text-xs"
                  >
                    Q1
                  </th>
                  <th
                    class="uppercase text-xs"
                  >
                    Q2
                  </th>
                  <th
                    class="uppercase text-xs"
                  >
                    Q3
                  </th>
                  <th
                    class="uppercase text-xs"
                  >
                    Q4
                  </th>
                  <th
                    align="right"
                    class="uppercase text-xs pr-4"
                  >
                    Final
                  </th>
                </tr>
              </thead>
              <tbody
                class="divide-y"
              >
                <tr>
                  <td
                    class="font-bold text-sm pt-1 pb-1 pl-4"
                  >
                    LAL
                  </td>
                  <td
                    align="center"
                    class="text-sm"
                  >
                    20
                  </td>
                  <td
                    align="center"
                    class="text-sm"
                  >
                    15
                  </td>
                  <td
                    align="center"
                    class="text-sm"
                  >
                    25
                  </td>
                  <td
                    align="center"
                    class="text-sm"
                  >
                    32
                  </td>
                  <td
                    align="right"
                    class="pr-4 text-sm"
                  >
                    92
                  </td>
                </tr>
                <tr>
                  <td
                    class="font-bold text-sm pt-1 pb-1 pl-4"
                  >
                    BOS
                  </td>
                  <td
                    align="center"
                    class="text-sm"
                  >
                    24
                  </td>
                  <td
                    align="center"
                    class="text-sm"
                  >
                    34
                  </td>
                  <td
                    align="center"
                    class="text-sm"
                  >
                    31
                  </td>
                  <td
                    align="center"
                    class="text-sm"
                  >
                    42
                  </td>
                  <td
                    align="right"
                    class="pr-4 text-sm"
                  >
                    131
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        class="md:col-span-4"
      >
        <div
          class="grid gap-3 divide-y pt-6 md:pl-6 md:pt-0"
        >
          <div>
            <p
              class="text-2xl font-bold uppercase"
            >
              Top Performers
            </p>
          </div>
          <div
            class="pt-5"
          >
            <div
              class="grid gap-4 md:grid-cols-2"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="flex flex-col items-center gap-1 w-[115px]"
                >
                  <div>
                    <div
                      class="w-[48px] h-[48px]"
                    >
                      <img
                        alt="name headshot"
                        class="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
                        loading="lazy"
                        src="https://cdn.nba.com/headshots/nba/latest/260x190/1629028.png"
                      />
                    </div>
                  </div>
                  <div>
                    <p
                      class="text-sm text-center"
                    >
                      Deandre Ayton
                    </p>
                    <p
                      class="text-xs text-gray-500 text-center"
                    >
                      PHX | #22 | C
                    </p>
                  </div>
                </div>
                <div
                  class="grid grid-cols-3 flex-grow"
                >
                  <div>
                    <div>
                      <p
                        class="text-center font-bold text-xl"
                      >
                        28
                      </p>
                      <p
                        class="text-center text-sm uppercase"
                      >
                        Pts
                      </p>
                    </div>
                  </div>
                  <div>
                    <div>
                      <p
                        class="text-center font-bold text-xl"
                      >
                        12
                      </p>
                      <p
                        class="text-center text-sm uppercase"
                      >
                        Reb
                      </p>
                    </div>
                  </div>
                  <div>
                    <div>
                      <p
                        class="text-center font-bold text-xl"
                      >
                        5
                      </p>
                      <p
                        class="text-center text-sm uppercase"
                      >
                        Ast
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex items-center"
              >
                <div
                  class="flex flex-col items-center gap-1 w-[115px]"
                >
                  <div>
                    <div
                      class="w-[48px] h-[48px]"
                    >
                      <img
                        alt="name headshot"
                        class="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
                        loading="lazy"
                        src="https://cdn.nba.com/headshots/nba/latest/260x190/1629627.png"
                      />
                    </div>
                  </div>
                  <div>
                    <p
                      class="text-sm text-center"
                    >
                      Zion Williamson
                    </p>
                    <p
                      class="text-xs text-gray-500 text-center"
                    >
                      NOP | #1 | F
                    </p>
                  </div>
                </div>
                <div
                  class="grid grid-cols-3 flex-grow"
                >
                  <div>
                    <div>
                      <p
                        class="text-center font-bold text-xl"
                      >
                        35
                      </p>
                      <p
                        class="text-center text-sm uppercase"
                      >
                        Pts
                      </p>
                    </div>
                  </div>
                  <div>
                    <div>
                      <p
                        class="text-center font-bold text-xl"
                      >
                        8
                      </p>
                      <p
                        class="text-center text-sm uppercase"
                      >
                        Reb
                      </p>
                    </div>
                  </div>
                  <div>
                    <div>
                      <p
                        class="text-center font-bold text-xl"
                      >
                        3
                      </p>
                      <p
                        class="text-center text-sm uppercase"
                      >
                        Ast
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
